### PR TITLE
rel to #13761: remove deprecated androidannotations, first steps into fixing waypoint-tests

### PR DESCRIPTION
**Pull request by eddiemuc**
rel to #13761: remove deprecated androidannotations, first steps into fixing waypoint-tests

### CODE REVIEW

**Comment by eddiemuc** [comment]
PR is not ready for merge imho

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractAddWaypointActivityTest.java [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
-import cgeo.geocaching.EditWaypointActivity_;
-
 public abstract class AbstractAddWaypointActivityTest extends AbstractWaypointActivityTest {
     @Override
     protected void setUp() throws Exception {
         super.setUp();
-        setActivityIntent(new EditWaypointActivity_.IntentBuilder_(getInstrumentation().getContext()).geocode(getCache().getGeocode()).get());
-        getActivity();
+//        setActivityIntent(new EditWaypointActivity.IntentBuilder_(getInstrumentation().getContext()).geocode(getCache().getGeocode()).get());
+//        getActivity();
```
Comment (bekuno) [inline main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractAddWaypointActivityTest.java:8]: Remove or reactivate it.

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractEditWaypointActivityTest.java [hunk 2]
```diff
@@ -12,9 +11,9 @@ protected void setUp() throws Exception {
         super.setUp();
         createWaypoint();
         getCache().addOrChangeWaypoint(waypoint, true);
-        final int waypointId = getCache().getWaypoints().get(0).getId();
-        setActivityIntent(new EditWaypointActivity_.IntentBuilder_(getInstrumentation().getContext()).geocode(getCache().getGeocode()).waypointId(waypointId).get());
-        getActivity();
+//        final int waypointId = getCache().getWaypoints().get(0).getId();
+//        setActivityIntent(new EditWaypointActivity.IntentBuilder_(getInstrumentation().getContext()).geocode(getCache().getGeocode()).waypointId(waypointId).get());
+//        getActivity();
```
Comment (bekuno) [inline main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractEditWaypointActivityTest.java:16]: Remove or reactivate it.

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java [hunk 3]
```diff
@@ -15,42 +26,79 @@
...
+    private Geocache testCache;
+
+    @Before
+    public void setUp() throws Exception {
+        testCache = CgeoTestUtils.createTestCache();
+        DataStore.saveCache(testCache, Collections.singleton(LoadFlags.SaveFlag.CACHE));
+
+//        super.setUp();
+//        setActivityIntent(new EditWaypointActivity_.IntentBuilder_(getInstrumentation().getContext()).geocode(getCache().getGeocode()).get());
+//        getActivity();
```
Comment (bekuno) [inline main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java:51]: Remove or reactivate it.

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java [hunk 4]
```diff
@@ -15,42 +26,79 @@
...
+    public void tearDown() {
+        CgeoTestUtils.removeCacheCompletely(testCache.getGeocode());
+    }
+
+    @Suppress // this will fail on CI since it does not have FINE-LOCATION granted
+    @Test
+    public void checkPermissions() {
+        assertThat(ContextCompat.checkSelfPermission(ApplicationProvider.getApplicationContext(), Manifest.permission.ACCESS_FINE_LOCATION))
+                .isEqualTo(PackageManager.PERMISSION_GRANTED);
+    }
```
Comment (bekuno) [inline main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java:65]: This test and more below are suppressed. Is it not possible to reactivate them?

**Review by bekuno — COMMENTED** [review]
The review is only based codewise.

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java (hunk 4 above)
Comment (eddiemuc) [reply main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java:65]: > This test and more below are suppressed. Is it not possible to reactivate them?

It is possible as soon as #13798 is resolved. This is why I don't want to delete the tests

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java (hunk 3 above)
Comment (eddiemuc) [reply main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java:51]: ok

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java (hunk 4 above)
Comment (eddiemuc) [reply main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AddWaypointActivityTest.java:65]: I have commented the tests accordingly

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractAddWaypointActivityTest.java (hunk 1 above)
Comment (eddiemuc) [reply main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractAddWaypointActivityTest.java:8]: This class will be deleted eventually after #13798 is resolved and I can fix the GUI tests for good. I keep it only as reference for later test refactoring. I will comment the class accordingly, hope it is then fine to keep it for now.

File: main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractEditWaypointActivityTest.java (hunk 2 above)
Comment (eddiemuc) [reply main/src/androidTest/java/cgeo/geocaching/activity/waypoint/AbstractEditWaypointActivityTest.java:16]: This class will be deleted eventually after #13798 is resolved and I can fix the GUI tests for good. I keep it only as reference for later test refactoring. I will comment the class accordingly, hope it is then fine to keep it for now.

**Pull request merged**